packages/contracts/contracts/L1/teleportr/TeleportrDeposit.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.9;
+
+import { Lib_AddressResolver } from "../../libraries/resolver/Lib_AddressResolver.sol";
+
+/**
+ * @title TeleportrDeposit
+ *
+ * Shout out to 0xclem for providing the inspiration for this contract:
+ * https://github.com/0xclem/teleportr/blob/main/contracts/BridgeDeposit.sol
+ */
+contract TeleportrDeposit is Lib_AddressResolver {
+    /// The minimum amount that be deposited in a receive.
+    uint256 public minDepositAmount;
+    /// The maximum amount that be deposited in a receive.
+    uint256 public maxDepositAmount;
+    /// The maximum balance the contract can hold after a receive.
+    uint256 public maxBalance;
+    /// The total number of successful deposits received.
+    uint256 public totalDeposits;
+
+    /**
+     * @notice Emitted any time the minimum deposit amount is set.
+     * @param previousAmount The previous minimum deposit amount.
+     * @param newAmount The new minimum deposit amount.
+     */
+    event MinDepositAmountSet(uint256 previousAmount, uint256 newAmount);
+
+    /**
+     * @notice Emitted any time the maximum deposit amount is set.
+     * @param previousAmount The previous maximum deposit amount.
+     * @param newAmount The new maximum deposit amount.
+     */
+    event MaxDepositAmountSet(uint256 previousAmount, uint256 newAmount);
+
+    /**
+     * @notice Emitted any time the contract maximum balance is set.
+     * @param previousBalance The previous maximum contract balance.
+     * @param newBalance The new maximum contract balance.
+     */
+    event MaxBalanceSet(uint256 previousBalance, uint256 newBalance);
+
+    /**
+     * @notice Emitted any time the balance is withdrawn by the owner.
+     * @param owner The current owner and recipient of the funds.
+     * @param balance The current contract balance paid to the owner.
+     */
+    event BalanceWithdrawn(address indexed owner, uint256 balance);
+
+    /**
+     * @notice Emitted any time a successful deposit is received.
+     * @param depositId A unique sequencer number identifying the deposit.
+     * @param emitter The sending address of the payer.
+     * @param amount The amount deposited by the payer.
+     */
+    event EtherReceived(
+        uint256 indexed depositId,
+        address indexed emitter,
+        uint256 amount
+    );
+
+    /**
+     * @notice Ensures that the caller is the owner stored in the address manager.
+     */
+    modifier isOwner() {
+        require(msg.sender == libAddressManager.owner(), "Caller is not owner");
+        _;
+    }
```

Comment:
Note that `libAddressManager.owner()` is a very sensitive address that we should only access for upgrades or occasionally modifying parameters. 

I'd suggest:
1. use this modifier on the setter functions
2. create a different role and auth modifier for the `withdrawBalance()` call, which we can use more often and feel OK about using in a hot wallet (as I assume we'll want to drain this contract on a semi-regular basis).